cfe_internal/enterprise/CFE_knowledge.cf

```diff
@@ -51,17 +51,43 @@ bundle agent cfe_internal_setup_knowledge
       "$(install_logs)" -> { "ENT-4506" }
         perms => mog("0600", "root", "root" );
 
-      "$(cfe_internal_hub_vars.docroot)"
-      comment => "All files in there should be at least 0460. Giving cfapache group full access",
-      handle => "cfe_internal_setup_knowledge_files_doc_root_2",
-      file_select => cfe_internal_docroot_perms,
-      perms => mog("0460", "root", $(def.cf_apache_group) ),
-      depth_search => recurse_exclude("inf");   # see exclude dirs in recurse_exclude() body
-
       "$(cfe_internal_hub_vars.docroot)/."
-      comment => "The top level docroot dir should be at least 0460 with root:cfapache",
+      comment => "The top level docroot needs to be readable and executable by the web server.",
       handle => "cfe_internal_setup_knowledge_dir_doc_root",
-      perms => mog("0460", "root", $(def.cf_apache_group) );
+      perms => mog("0550", "root", $(def.cf_apache_group) );
+
+      "$(cfe_internal_hub_vars.docroot)/vendor/." -> { "CFE-951" }
+        comment => "The vendor directory contains dependencies from the code ignitor framework",
+        handle => "cfe_internal_setup_knowledge_dir_doc_root_vendor_dirs",
+        depth_search => recurse_with_base("inf"),
+        file_select => dirs,
+        perms => mog("0450", "root", $(def.cf_apache_group) );
+
+      "$(cfe_internal_hub_vars.docroot)/vendor/." -> { "CFE-951" }
+        comment => "The vendor directory contains dependencies from the code ignitor framework",
+        handle => "cfe_internal_setup_knowledge_dir_doc_root_vendor_not_dir",
+        depth_search => recurse_with_base("inf"),
+        file_select => not_dir,
+        perms => mog("0440", "root", $(def.cf_apache_group) );
+
+      "$(cfe_internal_hub_vars.public_docroot)/themes/." -> { "CFE-951" }
+        comment => "The vendor directory contains dependencies from the code ignitor framework",
+        handle => "cfe_internal_setup_knowledge_dir_doc_root_public_themes_dirs",
+        depth_search => recurse_with_base("inf"),
+        file_select => dirs,
+        perms => mog("0450", "root", $(def.cf_apache_group) );
+
+      "$(cfe_internal_hub_vars.public_docroot)/."
+        comment => "The public dir in the docroot needs the to be executable by the webserver",
+        handle => "cfe_internal_setup_knowledge_dir_doc_root_public",
+        perms => mog("0450", "root", $(def.cf_apache_group) );
+
+      "$(cfe_internal_hub_vars.public_docroot)/images/." -> { "CFE-951" }
+        comment => "The vendor directory contains dependencies from the code ignitor framework",
```

Comment:
wrong comment? this is images not vendor dir right?

Reply:
ACK